File: templates/libgdx-game/android/src/main/java/com/playroom/runtime/android/AndroidLauncher.java
package com.playroom.runtime.android;

import android.os.Bundle;
import com.badlogic.gdx.backends.android.AndroidApplication;
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration;
import com.playroom.runtime.GameKitGame;

public class AndroidLauncher extends AndroidApplication {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        AndroidApplicationConfiguration config = new AndroidApplicationConfiguration();
        config.useImmersiveMode = true;
        PlayGamesServicesV2 services = new PlayGamesServicesV2(this);
        initialize(new GameKitGame(services), config);
    }
}

File: templates/libgdx-kmp-game/android/src/main/kotlin/com/playroom/runtime/android/AndroidLauncher.kt
package com.playroom.runtime.android

import android.os.Bundle
import com.badlogic.gdx.backends.android.AndroidApplication
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration
import com.playroom.runtime.GameKitGame

class AndroidLauncher : AndroidApplication() {
    private var playGamesServices: PlayGamesServicesV2? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val config = AndroidApplicationConfiguration().apply {
            useImmersiveMode = true
            useAccelerometer = false
            useCompass = false
        }
        playGamesServices = PlayGamesServicesV2(this).apply {
            init()
        }
        initialize(GameKitGame(playGamesServices), config)
    }
}
